weepickle-implicits/src/main/scala-3/com/rallyhealth/weepickle/v1/implicits/CaseClassTo.scala

```diff
@@ -124,6 +125,22 @@ trait CaseClassToPiece extends MacrosCommon :
       To.merge[T](readers: _*)
   }
 
+  /*
+   * Default string decoding for Scala 3 enums. Mirror should always be for the sealed trait generated by the
+   * enum, which is a Sum type. So any calls here with Product type mirrors are rejected as compilation errors.
+   */
+  inline def macroEnumTo[T](using m: Mirror.Of[T]): To[T] = inline m match {
+    case m: Mirror.ProductOf[T] =>
+      error("Enumeration macro used for non-sealed trait")
+
+    case m: Mirror.SumOf[T] =>
+      val valueOf: String => T = macros.enumValueOf[T]
```

Comment:
Is there runtime cost for creating the `String => T` function?

Reply:
compile time :)